packages/sitecore-jss/src/graphql/app-root-query.ts

```diff
@@ -65,12 +65,20 @@ export async function getAppRootId(
 
   debug.dictionary('fetching site root for %s %s', language, siteName);
 
-  const fetchResponse = await client.request<AppRootQueryResult>(appRootQuery, {
+  let fetchResponse = await client.request<AppRootQueryResult>(appRootQuery, {
     jssAppTemplateId: jssAppTemplateId || SitecoreTemplateId.JssApp,
     siteName,
     language,
   });
 
+  if (!fetchResponse?.layout?.homePage?.rootItem?.length) {
```

Comment:
We should add ` && language !== 'en'` (no need to try with en if we already did)